File: domain/src/main/java/cn/byronlab/weather/domain/usecase/InitializeAppUseCase.java
package cn.byronlab.weather.domain.usecase;

import cn.byronlab.weather.domain.model.Unit;
import cn.byronlab.weather.domain.repository.AppStartupRepository;
import cn.byronlab.weather.domain.result.DomainResult;

public final class InitializeAppUseCase {

    private final AppStartupRepository repository;

    public InitializeAppUseCase(AppStartupRepository repository) {
        this.repository = repository;
    }

    public DomainResult<Unit> execute() {
        return repository.initialize();
    }
}

File: domain/src/main/java/cn/byronlab/weather/domain/usecase/InitializeAppUseCase.kt
package cn.byronlab.weather.domain.usecase

import cn.byronlab.weather.domain.repository.AppStartupRepository
import cn.byronlab.weather.domain.result.DomainResult
import javax.inject.Inject

class InitializeAppUseCase @Inject constructor(
    private val repository: AppStartupRepository,
) {

    suspend operator fun invoke(): DomainResult<Unit> = repository.initialize()
}
